core/src/execution/datafusion/expressions/cast.rs

```diff
@@ -395,6 +399,58 @@ impl Cast {
         Ok(cast_array)
     }
 
+    fn cast_float64_to_decimal128(
+        array: &dyn Array,
+        precision: u8,
+        scale: i8,
+        eval_mode: EvalMode,
+    ) -> CometResult<ArrayRef> {
+        let input = array.as_any().downcast_ref::<Float64Array>().unwrap();
+        let mut cast_array = PrimitiveArray::<Decimal128Type>::builder(input.len());
+
+        let mul = (precision as f64).powi(scale as i32);
```

Comment:
Not sure I understand how this works. Can you add a comment?

Reply:
nice catch.. it was a bug.

```let mul = 10_f64.powi(scale as i32);``` taken from decimal.rs - arrow_cast.

Reply:
Makes _much_ more sense now ( I thought this might be some new Math :) )
The unit test didn't catch this because the precision in the test was 10! Perhaps we can add another unit test with some odd numbers.